ui/raidboss/data/05-shb/alliance/the_tower_at_paradigms_breach.ts

```diff
@@ -468,6 +477,144 @@ const triggerSet: TriggerSet<Data> = {
       suppressSeconds: 5,
       response: Responses.aoe(),
     },
+    // Deploy Armaments
+    //
+    // This attack has two variations and can be cast from two headings. The
+    // first variation causes two line AOEs to hit with the center being
+    // safe. The second variation causes a single line AOE to be cast in the
+    // middle, with the sides safe. It can either be cast horizontally or
+    // vertically over the arena.
+    //
+    // There are several skill IDs involved:
+    // 5C00: indicate start of a middle line attack. Always appears with one
+    //       5C02 cast
+    // 5C03: indicate start of a two side lines attack. Always appears with
+    //       2x 5C05 casts
+    // 5C01: indicates a single line attack comboing with the other boss.
+    //       Always appears simultaneously with the other bosses abilities
+    //       and a 6078 cast.
+    // 5C04: indicates a two side lines attack comboing with the other boss
+    //       Always appears simultaneously with the other bosses abilities
+    //       and 2x 6079 casts.
+    //
+    // To handle overlap, we have one trigger which captures all of the
+    // events into an array. Then, a delayed trigger processes all of the
+    // triggers and determines what to message players.
+    {
+      id: 'Paradigm Meng-Zi/Xun-Zi Deploy Armaments Collect',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['5C00', '5C01', '5C03', '5C04'] }),
+      run: (data, matches) => {
+        data.deployArmaments ??= [];
+        if (matches.id === '5C00' || matches.id === '5C01') {
+          if (matches.heading === '1.570772')
```

Comment:
Are they always in one of two spots?  I usually try to avoid matching exact floating point number strings.  Maybe they're the same now, but maybe they'll be slightly different in the future?

In the past for headings, I've done stuff like [this](https://github.com/quisquous/cactbot/blob/main/ui/raidboss/data/05-shb/eureka/delubrum_reginae_savage.ts#L255) to turn heading into a 0=N, 1=E, 2=S, 3=W integer.  I think that will be more robust in the future.

Reply:
They were always in these two spots for every log I checked. But I like the more robust pattern.

Reply:
Yeah, it's less about 2 vs 4 spots and more about not trusting a floating point string to be exactly that same string into the future.